### PR TITLE
Modify HTTPretty to allow for assertion testing

### PR DESCRIPTION
**Pull request by jamesfneyer**
When running pytests, there is not a very clear and easy way to test that all your mocked calls have been made, and if they have been made with the expected data. This provides a way to ensure that all the calls were made as expected, via passing in a list of assertions to HTTPretty.
This is run by using the `HTTPrettyAsserter` in a yield fixture, and passing in a list of assertions to the `assert_calls` function on the mocked calls.

### CODE REVIEW

File: tests/test_httpretty_asserter.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+def http_pretty():
+    httpretty.enable(allow_net_connect=False)
+    yield httpretty
+    httpretty.disable()
+
+
+@pytest.fixture
+def query_params():
+    query_params = {
+        'string': f'?query_param_1={1}&query_param_2={2}&query_param_bool={False}&datetime={CURRENT_TIME}',
```
[the remark this answers is not in the document]
Comment (ajhodges) [reply tests/test_httpretty_asserter.py]: @jamesfneyer

[the remark this answers is not in the document]
Comment (jamesfneyer) [reply tests/test_httpretty_asserter.py]: I originally didn't include that as using the parse_qs would've required the `parse_dict` functionality from the httpretty_asserter, but I guess I can include it.

[the remark this answers is not in the document]
Comment (ajhodges) [reply tests/test_httpretty_asserter.py]: I don't understand what you mean here. I am pretty confident that you can replace most of your parsing logic with those two urllib functions. You shouldn't need to manually parse URLs or querystrings.

**Comment by ajhodges** [comment]
Were you happy with this? I remember you said there were some changes you wanted to make to mine.

**Comment by jamesfneyer** [comment]
Thinking about it, I'm curious as to whether you believe that the path should always be required in the assertion? does it make sense to check that when there is no path?

**Comment by ajhodges** [comment]
What do you mean by "does it make sense to check that when there is no path"? I don't know how you could make HTTPretty assertions without a URL.

Also, it looks like the commits on this branch got screwed up somehow. I am seeing changes in the diff that are already in master. Please rebase.

**Comment by jamesfneyer** [comment]
Basically, the path is checking everything after the `.com` or whatever ending, so for things like calling on the url shortener we are just hitting `t.shipchain.io` with no path at the end and I was wondering if it made sense to still require a path check when it would be blank

**Comment by ajhodges** [comment]
If a request is made to `t.shipchain.io`, the path is `/`

**Comment by jamesfneyer** [comment]
And in those instances does it still make sense to require that as a parameter in the assertion? It couldn't hurt but I was just thinking that it might seem like an unnecessary check

**Comment by ajhodges** [comment]
Yes, assertions should be explicit

**Pull request merged**